### DISCUSSION TITLE
Rank-based User List

### DISCUSSION THREAD

**Discussion by ZooFology** [vertyco/vrt-cogs#101] [Ideas]
**Desired Behavior:**
I would like the member list to be organized based on activity, with the most active members at the top and the least active members at the bottom.

**Current Method:**
Currently achieved by creating many level roles that display separately in the member list. While this method works, it doesn't provide 100% accuracy top to bottom. Additionally, more roles will mean more accuracy, but it also creates too many separations.

**Proposed Feature:**
I propose implementing an optional setting that enables the display of users' ranks as prefixes in their nicknames. To ensure accuracy, the ranks should follow the formats below:

01 - for user count 1-99
001 - for user count 100-999
0001 - for user count 1000-9999

**Aesthetics Proposal:**
To reduce the footprint of the rank in a user's nickname, I propose using a Unicode font. Specifically, the Unicode subscript pseudoalphabet can be used. The small characters in this font will help maintain a clear distinction between the rank and the username.

0 = ₀
1 = ₁
2 = ₂
3 = ₃
4 = ₄
5 = ₅
6 = ₆
7 = ₇
8 = ₈
9 = ₉

[Click here for a visual Demonstration](https://i.imgur.com/H4W0OkK.png)

**Disclaimer:**
While this feature is of great importance to me, I understand if you consider it too much of a niche feature for your time. I would be satisfied with any of the following outcomes:

1. Implementing the feature into your current LevelUp cog for public use.
2. Implementing the feature as a separate cog, either publicly or privately.
3. Considering a paid commission for implementing outcome 1 or 2.

4. If you are unable to implement the feature, please allow me permission to post a bounty for someone else to create a cog that requests the LevelUp data. You have my full cooperation if the cog needs to be made public.

Thank you for considering this feature request. I believe that organizing the member list by activity will lead to an increase in server participation as well as encourage more interaction with the LevelUp Cog.

**Comment by vertyco** [comment]
Apologies if I misunderstood, but from what I can tell you want the cog to modify every user's nickname in the server and keep the rankings updated?
If that is indeed what you want, then this would require a large amount of API calls to the point of abusing it, which is against Discord ToS.

This does sound really cool, but I am not convinced it is a good idea due to the glaring reason I mentioned above.

**Reply by ZooFology** [reply]
Oh I see. That totally makes sense, especially if the feature is turned on and just starts bulk naming. Though I wonder if it would still be excessive if names were already set and the bot just needs to update existing and rename new users. I'm not trying to convince you of anything, I'm just genuinely curious if it is mainly all due to the bulk naming initially?
